target/linux/qualcommax/ipq50xx/base-files/lib/upgrade/platform.sh

```diff
@@ -72,6 +72,10 @@ platform_check_image() {
 
 platform_do_upgrade() {
 	case "$(board_name)" in
+	glinet,gl-b3000)
+		CI_UBIPART="rootfs"
```

Comment:
In OEM software volume is called `ubi_rootfs` not `rootfs`. Additionally we can remove unused volume with wifi firmware.
```suggestion
		CI_UBIPART="ubi_rootfs"
		remove_oem_ubi_volume wifi_fw
```

Reply:
The use of `CI_UBIPART="rootfs"` is incorrect. Why did you mark this as solved?

Reply:
because "ubi_rootfs" is only correct in oem firmware, were is essentially unchecked (doesn't matter),  This does matter in openwrt and changing this causes luci upgrad to fail. resulting in soft brick

Reply:
also, wifi partition is wiped during flash to openwrt. No need to remove

Reply:
also wifi partition is not retained through upgarde ... no need to remove it .

```
IPQ5018# smem                                                                   
ubi0: attaching mtd1                                                            
ubi0: scanning is finished                                                      
ubi0: attached mtd1 (name "mtd=0", size 120 MiB)                                
ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes                  
ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 2048                   
ubi0: VID header offset: 2048 (aligned 2048), data offset: 4096                 
ubi0: good PEBs: 960, bad PEBs: 0, corrupted PEBs: 0                            
ubi0: user volume: 3, internal volumes: 1, max. volumes count: 128              
ubi0: max/mean erase counter: 2/0, WL threshold: 4096, image sequence number: 11
ubi0: available PEBs: 0, total reserved PEBs: 960, PEBs reserved for bad PEB ha0
flash_type:             0xb                                                     
flash_index:            0x0                                                     
flash_chip_select:      0x0                                                     
flash_block_size:       0x20000                                                 
flash_density:          0x80000                                                 
partition table offset  0x0                                                     
No.: Name             Attributes            Start             Size              
  0: 0:SBL1           0x0000ffff              0x0          0x80000              
  1: 0:MIBIB          0x0000ffff          0x80000          0x80000              
  2: 0:BOOTCONFIG     0x0000ffff         0x100000          0x40000              
  3: 0:QSEE           0x0000ffff         0x140000         0x100000              
  4: 0:DEVCFG         0x0000ffff         0x240000          0x40000              
  5: 0:CDT            0x0000ffff         0x280000          0x40000              
  6: 0:APPSBLENV      0x0000ffff         0x2c0000          0x80000              
  7: 0:APPSBL         0x0000ffff         0x340000         0x140000              
  8: 0:ART            0x0000ffff         0x480000         0x100000              
  9: 0:TRAINING       0x0000ffff         0x580000          0x80000              
 10: CFG              0x0000ffff         0x600000         0x200000              
 11: rootfs           0x0000ffff         0x800000        0x7800000              
        ubi vol 0 kernel                                                        
        ubi vol 1 rootfs                                                        
        ubi vol 2 rootfs_data
```

Reply:
@testuser7 

Ok so some major progress. This convo had me curious as to why in stock ubi_rootfs is used for the volume name... so i dug a little deeper, it seems that this change is important, As i mentioned previously, I found how to switch uboot into nand mode, which seemed to have no effect until I made the change you suggested. To my surprise, it upgraded and loaded the kernel ... which crashes almost imidiately, as it tries to load the file system ...because .img is ubiFit not FIT.  So if you can hammer out the proper receipe to build the openwrt fit image, I can make a universal img for openwrt. I will also mention this to glinet devs in hopes that they'll release a FIT version of the oem. Then both oen and openwrt are universal from each webui ..luci, glient, and uboot   

more info to follow ....

Reply:
> So when updating via the web interface the entire partition is overwritten and not individual volumes? I thought that was only the case when updating via u-boot.

That is correct.  At least when using the .img file to upgrade, ubi is wiped

Reply:
> The use of `CI_UBIPART="rootfs"` is incorrect. Why did you mark this as solved?



> because "ubi_rootfs" is only correct in oem firmware, where it's essentially unchecked (doesn't matter), This does matter in openwrt and changing this causes luci upgrade to fail. resulting in soft brick

Still not entirely sure about this ...need more invertigation. Setting label name to ubi_rootfs seems to cause softbrick when upgrading with luci. But on the other hand, this can be use to enable nand mode in uboot.  Can we do a check for each and set accordingly ..


```
CI_UBIPART="rootfs"
if [ -n "$(find_mtd_chardev rootfs)"  ]; then 
            CI_UBIPART="ubi_rootfs"
fi
```

or a case .. something to that affect anyway??

Reply:
the above   won't work as is since find_mtd_chardev is looking at part name ... not label.  But I sure there's a way to check the label name.

Reply:
The correct one should be:
```
CI_UBIPART="rootfs"
CI_ROOTPART="ubi_rootfs"
```
Sorry for the confusion.